builds.go

```diff
@@ -0,0 +1,431 @@
+//
+// Copyright 2016, Arkbriar
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+)
+
+// BuildScope specifies five build status: pending, running, failed, success and canceled
+type BuildScope int
+
+const (
+	// BuildPending is one of five builds status.
+	BuildPending BuildScope = 1 << iota
+	// BuildRunning is one of five builds status.
+	BuildRunning
+	// BuildFailed is one of five builds status.
+	BuildFailed
+	// BuildSuccess is one of five builds status.
+	BuildSuccess
+	// BuildCanceled is one of five builds status.
+	BuildCanceled
+)
+
+func buildScope(scope BuildScope) string {
+	switch scope {
+	case BuildPending:
+		return "pending"
+	case BuildRunning:
+		return "running"
+	case BuildFailed:
+		return "failed"
+	case BuildSuccess:
+		return "success"
+	case BuildCanceled:
+		return "canceled"
+	default:
+		return "null"
+	}
+}
+
+// ListBuildsOptions are options for two list apis
+type ListBuildsOptions struct {
+	Scope *[]string `url:"scope, omitempty", json:"scope, omitempty"`
+}
+
+// BuildListBuildsOptions builds ListBuildsOptions from scopes in the form of integer.
+//
+// For example, BuildListBuildsOptions(BuildPending | BuildRunning | BuildFailed) will build
+// an option with "pending", "running" and "failed" scopes.
+func BuildListBuildsOptions(scopes int) *ListBuildsOptions {
+	opts := &ListBuildsOptions{
+		Scope: &[]string{},
+	}
+
+	if scopes == 0 {
+		return nil
+	}
+
+	for i := BuildPending; i <= BuildCanceled; i = i << 1 {
+		if (scopes & int(i)) != 0 {
+			temp := append(*opts.Scope, buildScope(i))
+			opts.Scope = &temp
+		}
+	}
+
+	if len(*opts.Scope) == 0 {
+		return nil
+	}
+
+	return opts
+}
+
+// BuildsService handles communication with the ci builds related methods
+// of the GitLab API.
+//
+// GitLab API docs: http://docs.gitlab.com/ce/api/builds.html
+type BuildsService struct {
+	client *Client
+}
+
+// Build represents a ci build.
+//
+// GitLab API docs: http://docs.gitlab.com/ce/api/builds.html
+type Build struct {
+	Commit *struct {
```

Comment:
I think we should use the existing `Commit` struct here: https://github.com/xanzy/go-gitlab/blob/master/commits.go#L36